app/src/components/player/index.tsx

```diff
@@ -8,26 +8,29 @@ import Description from "../typography/description";
 import Queue from "./queue";
 import Progress from "./progress";
 import PlaybackControls from "./playback-controls";
+import TrackPlayer, { useProgress, useTrackPlayerEvents, Event, State } from 'react-native-track-player';
 
 interface Props extends ViewProps {
-  queue: Record[];
   hideQueue?: boolean;
   hideDescription?: boolean;
 }
 
-const emptyQueueRecord: Record = { description: "-", duration: 0 };
+const Player: FC<Props> = ({ style, hideQueue, hideDescription }) => {
+  const [currentTrack, setCurrentTrack] = useState(null);
+  const [currentRecordIndex, setCurrentRecordIndex] = useState(0);
+  const [recordsCount, setRecordsCount] = useState(0);
+  const progress = useProgress();
+  const [isPlaying, setPlaying] = useState(false);
 
-const Player: FC<Props> = ({ style, queue, hideQueue, hideDescription }) => {
-  const recordsCount = queue.length;
-  const [currentRecordIndex, setCurrentRecordIndex] = useState(
-    recordsCount ? 0 : -1
-  );
-  const record = useMemo(
-    () => queue[currentRecordIndex] || emptyQueueRecord,
-    [currentRecordIndex, queue]
-  );
-
-  const [currentTime, setCurrentTime] = useState(0);
+  useTrackPlayerEvents([Event.PlaybackState], async (event) => {
+    const queue = await TrackPlayer.getQueue()
+    const currentIndex = await TrackPlayer.getPosition()
+    const state = await TrackPlayer.getState();
+    setRecordsCount(queue.length)
+    setCurrentRecordIndex(currentIndex)
+    setCurrentTrack(TrackPlayer.getTrack(currentIndex))
+    setPlaying(state == State.Playing)
```

Comment:
Jestli se všechny tyhle stavy nastavují společně, tak by bylo lepší mít setState jako object, než je volat (a re-renderovat) zvlášť

Reply:
Btw currentTrack se dá odvodit z currentIndex, takže bych ten state už neduplikoval (pokud teda TrackPlayer.getTrack není nějaká expensive operace, ale v tom případě by bylo lepší použít useMemo).

Reply:
Jednotlivé states mergnu do jednoho, to je jasné. Metoda `getTrack` volá nativní implementace na platformách. Díval jsem se jen na iOS a mimo to, že to hlídá existující index v queue tam nic speciálního není. 
Ten currentTrack bych si nechal.  Nebude se muset nic odvozovat a bude tam rovnou property -> čitelnější zápis pro budoucí generace 🌞